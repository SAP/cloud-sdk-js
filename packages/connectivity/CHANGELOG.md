# @sap-cloud-sdk/connectivity

## 2.5.0

### Patch Changes

- Updated dependencies [89f1c423]
  - @sap-cloud-sdk/util@2.5.0

## 2.4.0

### Minor Changes

- 04726a35: [Improvement] Reduce default log output on the `info` level significantly.
- 04726a35: [Compatibility Note] Only log the successful retrieval of destinations on the`info` log level, log everything else is on the `debug` or `warn` level.
- 0a008674: [Compatibility Note] Mark the function `noDestinationErrorMessage` as internal API.
- 0a008674: [New Functionality] Support self-signed certificate using the `trustStore` property of the destination object.

### Patch Changes

- Updated dependencies [0a008674]
  - @sap-cloud-sdk/util@2.4.0
