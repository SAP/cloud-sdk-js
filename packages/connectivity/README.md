<!-- sap-cloud-sdk-logo -->
<!-- This block is inserted by scripts/replace-common-readme.ts. Do not adjust it manually. -->
<a href="https://sap.com/s4sdk"><img src="https://help.sap.com/doc/2324e9c3b28748a4ae2ad08166d77675/1.0/en-US/logo-with-js.svg" alt="SAP Cloud SDK for JavaScript Logo" height="122.92" width="226.773"/></a>
<!-- sap-cloud-sdk-logo-stop -->

# @sap-cloud-sdk/connectivity

This package contains the connectivity functionality of the SAP Cloud SDK as well as the Cloud Platform abstractions.

## Installation

```
$ npm install @sap-cloud-sdk/connectivity
```

## Usage

The connectivity package is not intended for direct usage. 
The handles the platform abstractions for the connectivity e.g. connectivity service.

For more detailed overview visit our [OData client documentation](https://sap.github.io/cloud-sdk/docs/js/features/odata/use-odata-v2-type-safe-client-for-javascript-typescript).

<!-- sap-cloud-sdk-common-readme -->
<!-- This block is inserted by scripts/replace-common-readme.ts. Do not adjust it manually. -->
## Support

The recommended way to get in touch with us is to create an issue on [GitHub](https://github.com/SAP/cloud-sdk-js/issues).
Select the issue category `Bug`, `Feature` or `Question` depending on the nature of your request.
We try to provide fixes, features and answers as soon as possible.

## Contribute

If you would like to contribute to the SAP Cloud SDK, please make yourself familiar with our [contributing guidelines](https://github.com/SAP/cloud-sdk-js/blob/main/CONTRIBUTING.md) and follow the given instructions.

## Links

- [Github](https://github.com/SAP/cloud-sdk-js)
- [Github - Releases](https://github.com/SAP/cloud-sdk-js/releases)

<br>

- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk/)
- [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
- [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/docs/js/api-reference-js-ts)

<br>

- [developers.sap.com - Product Overview](https://developers.sap.com/topics/cloud-sdk.html)
- [developers.sap.com - Tutorials](https://developers.sap.com/tutorial-navigator.html?tag=products:technology-platform/sap-cloud-sdk/sap-cloud-sdk&tag=topic:javascript)

## License

The SAP Cloud SDK is released under the [Apache License Version 2.0.](http://www.apache.org/licenses/)
<!-- sap-cloud-sdk-common-readme-stop -->
