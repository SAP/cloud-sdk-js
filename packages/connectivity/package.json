{
  "name": "@sap-cloud-sdk/connectivity",
  "version": "2.5.0",
  "description": "SAP Cloud SDK for JavaScript connectivity",
  "homepage": "https://sap.github.io/cloud-sdk/docs/js/overview-cloud-sdk-for-javascript",
  "license": "Apache-2.0",
  "keywords": [
    "sap-cloud-sdk",
    "cloud-sdk",
    "sap-cloud-platform",
    "connectivity"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist/**/*.js",
    "dist/**/*.js.map",
    "dist/**/*.d.ts",
    "dist/**/*.d.ts.map",
    "internal.js",
    "internal.d.ts"
  ],
  "repository": "github:SAP/cloud-sdk-js",
  "scripts": {
    "compile": "tsc -b",
    "prepublishOnly": "yarn compile && yarn readme",
    "test": "yarn test:unit",
    "test:unit": "jest",
    "coverage": "jest --coverage",
    "lint": "eslint --ext .ts . && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -c",
    "lint:fix": "set TIMING=1 && eslint --ext .ts . --fix --quiet && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -w --loglevel error",
    "check:dependencies": "depcheck .",
    "check:public-api": "ts-node ../../scripts/check-public-api-cli.ts",
    "readme": "ts-node ../../scripts/replace-common-readme.ts"
  },
  "dependencies": {
    "@sap-cloud-sdk/util": "^2.5.0",
    "@sap/xsenv": "^3.3.1",
    "@sap/xssec": "^3.2.7",
    "axios": "^0.27.1",
    "http-proxy-agent": "^5.0.0",
    "https-proxy-agent": "^5.0.0",
    "jsonwebtoken": "^8.5.1",
    "opossum": "^6.0.0"
  },
  "devDependencies": {
    "@types/opossum": "^6.2.2",
    "base64url": "^3.0.1",
    "nock": "^13.2.6",
    "typescript": "~4.6.2"
  }
}
