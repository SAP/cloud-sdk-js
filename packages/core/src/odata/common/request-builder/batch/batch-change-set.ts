/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */

// TODO: Add specific request builder types as generics
/**
 * Representation of a batch change set, which holds a collection of write operations.
 */
export interface BatchChangeSet<T = any> {
  /**
   * Requests to combine to one change set.
   */
  readonly requests: T[];

  /**
   * Request boundary for separation of subrequests. Defaults to an autogenerated value.
   */
  readonly boundary: string;
}

// export function changeSet<T>(...requests: T[]): BatchChangeSet<T> {
//   return { requests };
// }

// export function batch<T>(
//   ...subRequests: (MethodRequestBuilderBase | BatchChangeSet<T>)[]
// ) {
//   return {
//     requests: subRequests
//   };
// }
