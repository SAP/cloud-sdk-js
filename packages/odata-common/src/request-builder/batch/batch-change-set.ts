import { v4 as uuid } from 'uuid';
import { DefaultDeSerializers, DeSerializers } from '../../de-serializers';
import { CreateRequestBuilderBase } from '../create-request-builder-base';
import { EntityBase } from '../../entity-base';
import { UpdateRequestBuilderBase } from '../update-request-builder-base';
import { DeleteRequestBuilderBase } from '../delete-request-builder-base';
import { ActionFunctionImportRequestBuilderBase } from '../action-function-import-request-builder-base';

/**
 * Representation of a batch change set, which holds a collection of write operations.
 */
export class BatchChangeSet<
  DeSerializersT extends DeSerializers = DefaultDeSerializers
> {
  /**
   * Create an instance of BatchChangeSet.
   * @param requests - Requests to combine to one change set.
   * @param boundary - Request boundary for separation of sub requests. Defaults to an auto generated value.
   */
  constructor(
    readonly requests: ChangesetBuilderTypes<DeSerializersT>[],
    readonly boundary: string = `changeset_${uuid()}`
  ) {}
}

/**
 * Some function imports contain not serializable entities and the execute() method is removed from them.
 * Since the execute method is not needed in batch the execute it is removed from all function imports.
 */
export type ChangesetBuilderTypes<DeSerializersT extends DeSerializers> =
  | CreateRequestBuilderBase<EntityBase, DeSerializersT>
  | UpdateRequestBuilderBase<EntityBase, DeSerializersT>
  | DeleteRequestBuilderBase<EntityBase, DeSerializersT>
  | Omit<ActionFunctionImportRequestBuilderBase<any, any>, 'execute'>;
