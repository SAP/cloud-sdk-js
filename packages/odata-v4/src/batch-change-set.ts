import { v4 as uuid } from 'uuid';
import { BatchChangeSet } from '@sap-cloud-sdk/odata-common';
import { Entity } from './entity';
import { CreateRequestBuilder } from './request-builder/create-request-builder';
import { UpdateRequestBuilder } from './request-builder/update-request-builder';
import { DeleteRequestBuilder } from './request-builder/delete-request-builder';

/**
 * @deprecated Since v1.30.0. Use [[BatchChangeSet]] directly
 * Representation of a batch change set, which holds a collection of write operations.
 */
export class ODataBatchChangeSet<
  RequestT extends
    | CreateRequestBuilder<Entity>
    | UpdateRequestBuilder<Entity>
    | DeleteRequestBuilder<Entity>
> implements BatchChangeSet<RequestT>
{
  /**
   * @deprecated Since v1.30.0. Use [[boundary]] instead.
   */
  get changeSetId(): string {
    return this.boundary;
  }

  /**
   * Create an instance of ODataBatchChangeSet.
   * @param requests - Requests to combine to one change set.
   * @param boundary - Boundary used in the multipart request.
   */
  constructor(
    readonly requests: RequestT[],
    readonly boundary: string = `changeset_${uuid()}`
  ) {}
}
