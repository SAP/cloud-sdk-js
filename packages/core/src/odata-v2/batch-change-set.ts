/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
import { v4 as uuid } from 'uuid';
import { BatchChangeSet } from '../odata-common/request-builder/batch/batch-change-set';
import { Entity } from './entity';
import {
  CreateRequestBuilder,
  DeleteRequestBuilder,
  UpdateRequestBuilder
} from './request-builder';

/**
 * @deprecated Since v1.30.0. Use [[BatchChangeSet]] directly
 * Representation of a batch change set, which holds a collection of write operations.
 */
export class ODataBatchChangeSet<
  RequestT extends
    | CreateRequestBuilder<Entity>
    | UpdateRequestBuilder<Entity>
    | DeleteRequestBuilder<Entity>
> implements BatchChangeSet<RequestT> {
  /**
   * @deprecated Since v1.30.0. Use [[boundary]] instead.
   */
  get changeSetId() {
    return this.boundary;
  }

  /**
   * Create an instance of ODataBatchChangeSet.
   * @param requests Requests to combine to one change set.
   * @param boundary Boundary used in the multipart request.
   */
  constructor(
    readonly requests: RequestT[],
    readonly boundary: string = `changeset_${uuid()}`
  ) {}
}

export { ODataBatchChangeSet as ODataBatchChangeSetV2 };
