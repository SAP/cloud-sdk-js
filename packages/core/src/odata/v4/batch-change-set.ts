/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
import { v4 as uuid } from 'uuid';
import { BatchChangeSet } from '../common/request-builder/batch/batch-change-set';
import { EntityV4 } from './entity';
import {
  CreateRequestBuilderV4,
  DeleteRequestBuilderV4,
  UpdateRequestBuilderV4
} from './request-builder';

/**
 * @deprecated Since v1.30.0. Use [[BatchChangeSet]] directly
 * Representation of a batch change set, which holds a collection of write operations.
 */
export class ODataBatchChangeSetV4<
  RequestT extends
    | CreateRequestBuilderV4<EntityV4>
    | UpdateRequestBuilderV4<EntityV4>
    | DeleteRequestBuilderV4<EntityV4>
> implements BatchChangeSet<RequestT> {
  /**
   * @deprecated Since v1.30.0. Use [[boundary]] instead.
   */
  get changeSetId() {
    return this.boundary;
  }

  /**
   * Create an instance of ODataBatchChangeSetV4.
   * @param requests Requests to combine to one change set.
   * @param boundary Boundary used in the multipart request.
   */
  constructor(
    readonly requests: RequestT[],
    readonly boundary: string = `changeset_${uuid()}`
  ) {}
}
