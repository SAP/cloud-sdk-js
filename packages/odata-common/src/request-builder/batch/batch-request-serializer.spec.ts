import { Destination } from '@sap-cloud-sdk/connectivity';
import {
  CommonEntity,
  commonEntityApi,
  CommonEntityApi
} from '@sap-cloud-sdk/test-services-odata-common/common-entity';
import {
  batchRequestBuilder,
  createRequestBuilder,
  deleteRequestBuilder,
  getAllRequestBuilder,
  getByKeyRequestBuilder,
  updateRequestBuilder,
  WriteBuilder
} from '@sap-cloud-sdk/test-services-odata-common/common-request-config';
import {
  serializeBatchRequest,
  serializeRequest,
  serializeChangeSet
} from './batch-request-serializer';
import { BatchChangeSet } from './batch-change-set';

jest.mock('uuid', () => ({
  v4: jest.fn(() => '<content-id>')
}));

function commonEntity(): CommonEntity {
  return new CommonEntityApi()
    .entityBuilder()
    .keyPropertyGuid('guidId')
    .keyPropertyString('strId')
    .stringProperty('value')
    .build();
}
const entityKeys = {
  KeyPropertyGuid: 'testId',
  KeyPropertyString: 'test'
};

function changeSet(requests: WriteBuilder[]): BatchChangeSet {
  return new BatchChangeSet<any>(requests, 'changeSet_boundary');
}

describe('batch request serializer', () => {
  describe('serializeRequest', () => {
    it('serializes getAll request', () => {
      expect(serializeRequest(getAllRequestBuilder())).toMatchSnapshot();
    });

    it('serializes getAll request with filter', () => {
      expect(
        serializeRequest(
          getAllRequestBuilder({
            filter: commonEntityApi.schema.STRING_PROPERTY.equals('test')
          })
        )
      ).toMatchSnapshot();
    });

    it('encodes user provided filter parameters only once', () => {
      expect(
        serializeRequest(
          getAllRequestBuilder({
            filter:
              commonEntityApi.schema.STRING_PROPERTY.equals('with EmptySpace')
          })
        )
      ).toMatch(/filter=\(StringProperty%20eq%20'with%20EmptySpace'\)/);
    });

    it('serializes getAll request with custom headers', () => {
      expect(
        serializeRequest(
          getAllRequestBuilder({ headers: { 'Custom-Header': 'custom' } })
        )
      ).toMatchSnapshot();
    });

    it('serializes getAll request with absolute sub request path', () => {
      expect(
        serializeRequest(getAllRequestBuilder(), {
          subRequestPathType: 'absolute',
          destination: { url: 'http://example.com' }
        })
      ).toMatchSnapshot();
    });

    it('serializes getAll request with entity relative sub request path', () => {
      expect(
        serializeRequest(getAllRequestBuilder(), {
          subRequestPathType: 'relativeToEntity',
          destination: { url: 'http://example.com' }
        })
      ).toMatchSnapshot();
    });

    it('serializes getByKey request', () => {
      expect(
        serializeRequest(getByKeyRequestBuilder({ keys: entityKeys }))
      ).toMatchSnapshot();
    });

    it('serializes create request', () => {
      expect(
        serializeRequest(createRequestBuilder({ payload: commonEntity() }))
      ).toMatchSnapshot();
    });

    it('serializes update request', () => {
      const test = serializeRequest(
        updateRequestBuilder({ payload: commonEntity() })
      );
      expect(test).toMatchSnapshot();
    });

    it('serializes update request using put', () => {
      const updateRequest = updateRequestBuilder({
        payload: commonEntity()
      }).replaceWholeEntityWithPut();
      expect(serializeRequest(updateRequest)).toMatchSnapshot();
    });

    it('serializes delete request with entity', () => {
      expect(
        serializeRequest(deleteRequestBuilder({ payload: commonEntity() }))
      ).toMatchSnapshot();
    });

    it('serializes delete request with id', () => {
      expect(
        serializeRequest(deleteRequestBuilder({ keys: entityKeys }))
      ).toMatchSnapshot();
    });

    it('serializes delete request with eTag', () => {
      const withEtag = commonEntity().setVersionIdentifier('eTag');
      expect(
        serializeRequest(deleteRequestBuilder({ payload: withEtag }))
      ).toMatchSnapshot();
    });
  });

  describe('serializeChangeSet', () => {
    it('serializes change set with one operation', () => {
      const createRequest = createRequestBuilder({ payload: commonEntity() });
      expect(serializeChangeSet(changeSet([createRequest]))).toMatchSnapshot();
    });

    it('serializes change set with multiple operations', () => {
      const updateRequest = updateRequestBuilder({ payload: commonEntity() });
      const createRequest = createRequestBuilder({ payload: commonEntity() });
      expect(
        serializeChangeSet(
          new BatchChangeSet<any>(
            [updateRequest, createRequest],
            'changeSet_boundary'
          )
        )
      ).toMatchSnapshot();
    });

    it('returns undefined for empty change set', () => {
      expect(serializeChangeSet(changeSet([]))).toBeUndefined();
    });
  });

  describe('serializeBatchRequest', () => {
    it('serializes payload for batch subrequests', () => {
      const requests = [
        changeSet([createRequestBuilder({ payload: commonEntity() })]),
        getAllRequestBuilder(),
        changeSet([
          updateRequestBuilder({ payload: commonEntity() }),
          deleteRequestBuilder({ keys: entityKeys })
        ]),
        getByKeyRequestBuilder({ keys: entityKeys })
      ];
      expect(
        serializeBatchRequest(batchRequestBuilder(requests))
      ).toMatchSnapshot();
    });

    it("throws an error if the request option 'absolute' is with a destination without url.", () => {
      const request = batchRequestBuilder([]).withSubRequestPathType(
        'absolute'
      );
      expect(() =>
        serializeBatchRequest(request, {
          subRequestPathType: request.requestConfig.subRequestPathType,
          destination: {} as Destination
        })
      ).toThrowErrorMatchingInlineSnapshot(
        '"Cannot serialize batch request. Invalid destination provided for sub request path type \'absolute\'"'
      );
    });
  });
});
