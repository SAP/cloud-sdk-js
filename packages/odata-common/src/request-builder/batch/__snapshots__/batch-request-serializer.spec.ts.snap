// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`batch request serializer serializeBatchRequest serializes payload for batch subrequests 1`] = `
"--batch_fixed_boundary_for_testing
Content-Type: multipart/mixed; boundary=changeSet_boundary

--changeSet_boundary
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

POST /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"KeyPropertyGuid\\":\\"guidId\\",\\"KeyPropertyString\\":\\"strId\\",\\"StringProperty\\":\\"value\\"}

--changeSet_boundary--
--batch_fixed_boundary_for_testing
Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json


--batch_fixed_boundary_for_testing
Content-Type: multipart/mixed; boundary=changeSet_boundary

--changeSet_boundary
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

PATCH /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=guidId,KeyPropertyString='strId') HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"StringProperty\\":\\"value\\"}

--changeSet_boundary
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

DELETE /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=testId,KeyPropertyString='test') HTTP/1.1
Content-Type: application/json
Accept: application/json



--changeSet_boundary--
--batch_fixed_boundary_for_testing
Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=testId,KeyPropertyString='test') HTTP/1.1
Content-Type: application/json
Accept: application/json


--batch_fixed_boundary_for_testing--
"
`;

exports[`batch request serializer serializeChangeSet serializes change set with multiple operations 1`] = `
"Content-Type: multipart/mixed; boundary=changeSet_boundary

--changeSet_boundary
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

PATCH /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=guidId,KeyPropertyString='strId') HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"StringProperty\\":\\"value\\"}

--changeSet_boundary
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

POST /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"KeyPropertyGuid\\":\\"guidId\\",\\"KeyPropertyString\\":\\"strId\\",\\"StringProperty\\":\\"value\\"}

--changeSet_boundary--"
`;

exports[`batch request serializer serializeChangeSet serializes change set with one operation 1`] = `
"Content-Type: multipart/mixed; boundary=changeSet_boundary

--changeSet_boundary
Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

POST /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"KeyPropertyGuid\\":\\"guidId\\",\\"KeyPropertyString\\":\\"strId\\",\\"StringProperty\\":\\"value\\"}

--changeSet_boundary--"
`;

exports[`batch request serializer serializeRequest serializes create request 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

POST /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"KeyPropertyGuid\\":\\"guidId\\",\\"KeyPropertyString\\":\\"strId\\",\\"StringProperty\\":\\"value\\"}
"
`;

exports[`batch request serializer serializeRequest serializes delete request with eTag 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

DELETE /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=guidId,KeyPropertyString='strId') HTTP/1.1
Content-Type: application/json
Accept: application/json
Some-Implementation-For-Test: eTag


"
`;

exports[`batch request serializer serializeRequest serializes delete request with entity 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

DELETE /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=guidId,KeyPropertyString='strId') HTTP/1.1
Content-Type: application/json
Accept: application/json


"
`;

exports[`batch request serializer serializeRequest serializes delete request with id 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

DELETE /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=testId,KeyPropertyString='test') HTTP/1.1
Content-Type: application/json
Accept: application/json


"
`;

exports[`batch request serializer serializeRequest serializes getAll request 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

"
`;

exports[`batch request serializer serializeRequest serializes getAll request with absolute sub request path 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary

GET http://example.com/sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

"
`;

exports[`batch request serializer serializeRequest serializes getAll request with custom headers 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json
Custom-Header: custom

"
`;

exports[`batch request serializer serializeRequest serializes getAll request with entity relative sub request path 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary

GET /A_CommonEntity HTTP/1.1
Content-Type: application/json
Accept: application/json

"
`;

exports[`batch request serializer serializeRequest serializes getAll request with filter 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity?$filter=(StringProperty%20eq%20'test') HTTP/1.1
Content-Type: application/json
Accept: application/json

"
`;

exports[`batch request serializer serializeRequest serializes getByKey request 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=testId,KeyPropertyString='test') HTTP/1.1
Content-Type: application/json
Accept: application/json

"
`;

exports[`batch request serializer serializeRequest serializes update request 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

PATCH /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=guidId,KeyPropertyString='strId') HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"StringProperty\\":\\"value\\"}
"
`;

exports[`batch request serializer serializeRequest serializes update request using put 1`] = `
"Content-Type: application/http
Content-Transfer-Encoding: binary
Content-Id: <content-id>

PUT /sap/opu/odata/sap/API_COMMON_SRV/A_CommonEntity(KeyPropertyGuid=guidId,KeyPropertyString='strId') HTTP/1.1
Content-Type: application/json
Accept: application/json

{\\"KeyPropertyGuid\\":\\"guidId\\",\\"KeyPropertyString\\":\\"strId\\",\\"StringProperty\\":\\"value\\"}
"
`;
